tests/sources/python-modules.py

```diff
@@ -248,6 +248,11 @@ def replace(lst, old, new):
         'dataclasses',
     ])
 
+if (sys.version_info.major, sys.version_info.minor, sys.version_info.micro) == (3, 7, 17):
+    standard_library.remove('bz2')
+    standard_library.remove('curses')
+    standard_library.remove('readline')
```

Comment:
This might have made your tests pass, but it's not a fix as it ignores a real build issue with Python 3.7.17: 
https://github.com/actions/setup-python/issues/682.

Note that 3.7.17's `build_output.txt` logs that the build setup was unable to find the "necessary bits" to build these Python standard libraries:

```
Python build finished successfully!
The necessary bits to build these optional modules were not found:
_bz2                  _curses               _curses_panel      
ossaudiodev           readline              spwd               
To find the necessary bits, look in setup.py in detect_modules() for the module's name.
```

The real fix is addressing those missing dependencies ("necessary bits") so the Python standard library modules can be built.

All of the other `standard_library.remove(…)` lines in this file are for legitimate upstream removals of modules from Python's standard lib, not ignoring of local build misconfigurations.